### PR TITLE
Header fields and request method support

### PR DESCRIPTION
**Pull request by fitterman**
This pull request includes many new capabilities. The basic approach was to consolidate the GET and POST methods into a single base method that exposes configuration options and to turn the current public methods into convenience methods that access the merged method.

New parameters allow control over the request method, signature method, HTTP headers and how the data is transmitted (form or JSON in the body). Automated tests were added as well.

### CODE REVIEW

**Comment by mxcl** [comment]
Marvelous, I'll review today.

**Comment by mxcl** [comment]
I'd prefer that the testing hooks were compile time (that is, #defines). I assume it would be possible?

Otherwise looks great, couldn't see any obvious regressions. This many changes are scary however. Glad you added tests.

**Comment by mxcl** [comment]
Probably I'll remove the hardcoded drop in of OMGHTTPURLRQ also but post merge.

**Comment by fitterman** [comment]
I can change the testing hooks to #defines. I'd originally thought about using a Podfile instead of the hardcoded OMGHTTPURLRQ dropin. Do you want me to make that change as well? I was concerned it might confuse someone, but probably most people won't ever look in the test directory.

**Comment by fitterman** [comment]
I'm not sure about how to test xauth or what your release-testing migration plan is for this. If you can comment on that, I'd like to hear a bit as I don't have a heavy-duty app to test this with.

**Comment by mxcl** [comment]
I usually feel that if the tests don't work out-the-box, then that's fine.

File: TDOAuth.m [hunk 1]
```diff
@@ -78,13 +88,25 @@ - (id)initWithConsumerKey:(NSString *)consumerKey
            consumerSecret:(NSString *)consumerSecret
               accessToken:(NSString *)accessToken
               tokenSecret:(NSString *)tokenSecret
+          signatureMethod:(TDOAuthSignatureMethod)signatureMethod
 {
+    NSString *smString;
+    if (signatureMethod == TDOAuthSignatureMethodHmacSha256) {
+        smString = @"HMAC-SHA256";
+    } else if ((signatureMethod == TDOAuthSignatureMethodHmacSha1) || (!signatureMethod)) {
```
Comment (mxcl) [inline TDOAuth.m]: Unless I'm mistaken, the second part of this if statement is redundant. Since `TDOAuthSignatureMethodHmacSha1` is `0`.

Comment (fitterman) [inline TDOAuth.m]: Correct. This was left over from an earlier version when I was still passing a string in to specify the signature method. It's been removed.

**Pull request merged**